File: src/leetcode/PoorPigs.py
class Solution(object):
    def poorPigs(self, buckets, minutesToDie, minutesToTest):
        """
        :type buckets: int
        :type minutesToDie: int
        :type minutesToTest: int
        :rtype: int
        """
        numberOfPigs = 0
        while (minutesToTest / minutesToDie + 1) ** numberOfPigs < buckets:
            numberOfPigs += 1
        return numberOfPigs


if __name__ == "__main__":
    buckets = 1000
    minutesToDie = 15
    minutesToTest = 60
    solution = Solution()
    numberOfPigs = solution.poorPigs(buckets, minutesToDie, minutesToTest)
    print numberOfPigs

File: src/leetcode/PoorPigs.java
package leetcode;

public class PoorPigs {
    public static int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
    	int numberOfPigs = 0;
    	while (Math.pow(minutesToTest / minutesToDie + 1, numberOfPigs) < buckets) {
    	    numberOfPigs += 1;
    	}
	    return numberOfPigs;
	}
    
    public static void main(String[] args) {
    	int buckets = 1000;
    	int minutesToDie = 15;
    	int minutesToTest = 60;
    	int result = poorPigs(buckets, minutesToDie, minutesToTest);
    	System.out.print(result);
    }
}
